packages/ipfs-core-utils/src/cid.js

```diff
@@ -10,23 +10,22 @@ const CID = require('cids')
  * that it will also not apply the encoding (since v0 CIDs can only be encoded
  * as base58btc).
  *
- * @param {CID | Buffer | string} cid - The CID to encode
+ * @param {CID|Buffer|string} input - The CID to encode
  * @param {Object} [options] - Optional options
  * @param {string} [options.base] - Name of multibase codec to encode the CID with
  * @param {boolean} [options.upgrade] - Automatically upgrade v0 CIDs to v1 when
  * necessary. Default: true.
- * @returns {string}
+ * @returns {string} - CID in string representation
  */
-exports.cidToString = (cid, options) => {
-  options = options || {}
-  options.upgrade = options.upgrade !== false
-
-  if (!CID.isCID(cid)) {
-    cid = new CID(cid)
-  }
+exports.cidToString = (input, options = {}) => {
+  const upgrade = options.upgrade !== false
+  let cid = CID.isCID(input)
+    ? input
+    // @ts-ignore - TS seems to get confused by the type defs in CID repo.
```

Comment:
Is there a PR open to fix this?

Reply:
No there is no PR but I opened an issue https://github.com/multiformats/js-cid/issues/128

I would like to get this inn and then go over all the ts-ignores and address them as followups.